sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -36,20 +36,29 @@ internal static List<TelemetryItem> OtelToAzureMonitorTrace(Batch<Activity> batc
                     var activityTagsProcessor = EnumerateActivityTags(activity);
                     telemetryItem = new TelemetryItem(activity, ref activityTagsProcessor, azureMonitorResource, instrumentationKey);
 
-                    // Check for Exceptions events
-                    if (activity.Events.Any())
-                    {
-                        AddTelemetryFromActivityEvents(activity, telemetryItem, telemetryItems);
-                    }
-
                     switch (activity.GetTelemetryType())
                     {
                         case TelemetryType.Request:
-                            telemetryItem.Data = new MonitorBase
+                            if (activity.Kind == ActivityKind.Server)
                             {
-                                BaseType = "RequestData",
-                                BaseData = new RequestData(Version, activity, ref activityTagsProcessor)
-                            };
+                                var (requestUrl, operationName) = GetHttpOperationNameAndUrl(activity.DisplayName, activityTagsProcessor.activityType, ref activityTagsProcessor.MappedTags);
+                                telemetryItem.Tags[ContextTagKeys.AiOperationName.ToString()] = operationName;
+                                telemetryItem.Tags[ContextTagKeys.AiLocationIp.ToString()] = TraceHelper.GetLocationIp(ref activityTagsProcessor.MappedTags);
```

Comment:
@vishweshbankwar I think for new HTTP semantics, we need to add `client.address`. We could follow up on a new PR.

Reply:
Also [UserAgent](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#:~:text=Recommended-,user_agent.original,-string)

Reply:
@TimothyMothra Is there a plan to add `client.address` to `AspNetCoreInstrumentation` library?

Reply:
There is no "plan" at the moment.
When working on the instrumentation library, I only converted the attributes that were already in use. I didn't add any new attributes.

The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions) describes `client.address` as Recommended, but not Required.
Lets discuss this in standup.